driver-core/src/main/com/mongodb/client/model/bulk/ClientWriteModel.java

```diff
@@ -0,0 +1,355 @@
+ */
+package com.mongodb.client.model.bulk;
+
+import com.mongodb.MongoNamespace;
+import com.mongodb.annotations.Sealed;
+import com.mongodb.client.model.Filters;
+import com.mongodb.client.model.Updates;
+import com.mongodb.internal.client.model.bulk.ClientDeleteManyModel;
+import com.mongodb.internal.client.model.bulk.ClientDeleteOneModel;
+import com.mongodb.internal.client.model.bulk.ClientInsertOneModel;
+import com.mongodb.internal.client.model.bulk.ClientReplaceOneModel;
+import com.mongodb.internal.client.model.bulk.ClientUpdateManyModel;
+import com.mongodb.internal.client.model.bulk.ClientUpdateOneModel;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * An individual write operation to be executed as part of a client-level bulk write operation.
+ *
+ * @since 5.3
+ */
+@Sealed
+public interface ClientWriteModel {
+    /**
+     * Creates a model for inserting the {@code document} into the {@code namespace}.
+     *
+     * @param namespace The namespace.
+     * @param document The document.
+     * @return The requested model.
+     * @param <TDocument> The document type, for example {@link Document}.
+     * @see Filters
+     */
+    static <TDocument> ClientWriteModel insertOne(
+            final MongoNamespace namespace,
+            final TDocument document) {
+        notNull("namespace", namespace);
+        notNull("document", document);
+        return new ClientInsertOneModel(namespace, document);
+    }
+
+    /**
+     * Creates a model for updating at most one document in the {@code namespace} that matches the {@code filter}.
+     * This method is functionally equivalent to {@link #updateOne(MongoNamespace, Bson, Bson, ClientUpdateOptions)}
+     * with the {@linkplain ClientUpdateOptions#clientUpdateOptions() default options}.
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param update The update.
+     * @return The requested model.
+     * @see Filters
+     * @see Updates
+     */
+    static ClientWriteModel updateOne(
+            final MongoNamespace namespace,
+            final Bson filter,
+            final Bson update) {
+        notNull("namespace", namespace);
+        notNull("filter", filter);
+        notNull("update", update);
+        return new ClientUpdateOneModel(namespace, filter, update, null, null);
+    }
+
+    /**
+     * Creates a model for updating at most one document in the {@code namespace} that matches the {@code filter}.
+     *
+     * @param namespace The namespace.
+     * @param filter The filter.
+     * @param update The update.
+     * @param options The options.
+     * @return The requested model.
+     * @see Filters
+     * @see Updates
+     */
+    static ClientWriteModel updateOne(
+            final MongoNamespace namespace,
+            final Bson filter,
+            final Bson update,
+            final ClientUpdateOptions options) {
```

Comment:
I'd prefer to half the number of overloads of methods in `ClientWriteModel` and the number of new `MongoCluster.bulkWrite` overloads by allowing to pass `@Nullable` options. However, the existing bulk write API does not allow `null` options, and I followed that in the new API.